### PR TITLE
add e2e test when cluster join, update, karmada need to reconcile search cache

### LINKED ISSUE #3731: Add E2E for #3670: when cluster update, karmada need to reconcile search cache
**Task description**:

#3670 make a patch when cluster update, Karmada will reconcile search cache. And then, we can add an E2E case to cover this update. 

We can add the case here:

https://github.com/karmada-io/karmada/blob/c39856bfe9a40bdc20a44cedb22c8fa5c4a5ceec/test/e2e/search_test.go#L343

**Who can join or take the task**:

Anyone who is interested can come and claim this job.

**How to join or take the task**:

Just reply on the issue with the message `/assign` in a separate line.

Then, the issue will be assigned to you.

**How to ask for help**:

If you need help or have questions, please feel free to ask on this issue.
The issue author or other members of the community will guide you through the contribution process.

### PR DESCRIPTION
**Pull request by zhy76**
**What type of PR is this?**

**What this PR does / why we need it**:
add e2e test when cluster join, update, karmada need to reconcile search cache
**Which issue(s) this PR fixes**:
Fixes #3731 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: test/e2e/search_test.go [hunk 1]
```diff
@@ -711,4 +717,197 @@ var _ = ginkgo.Describe("[karmada-search] karmada search testing", ginkgo.Ordere
...
+				clusterContext string
+				f              cmdutil.Factory
+
+				rrName           string
+				rr               *searchv1alpha1.ResourceRegistry
+				resourceSelector searchv1alpha1.ResourceSelector
+			)
+
+			ginkgo.BeforeAll(func() {
+				clusterName = "member-e2e-" + rand.String(3)
```
Comment (jwcesign) [inline test/e2e/search_test.go:738]: clusterName = "member-e2e-" + rand.String(RandomStrLength)

File: test/e2e/search_test.go [hunk 2]
```diff
@@ -711,4 +717,197 @@ var _ = ginkgo.Describe("[karmada-search] karmada search testing", ginkgo.Ordere
...
+				ginkgo.By(fmt.Sprintf("Deleting clusters: %s", clusterName), func() {
+					err := deleteCluster(clusterName, kubeConfigPath)
+					gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+					_ = os.Remove(kubeConfigPath)
+				})
+			})
+
+			ginkgo.It("[member clusters joined] could reconcile ResourceRegistry", func() {
+				// check the initial state of the ResourceRegistry
+				rr_test := framework.GetResourceRegistry(karmadaClient, rrName)
```
Comment (jwcesign) [inline test/e2e/search_test.go:789]: Keep the naming style consistent, so `rrTest`

**Review by jwcesign — COMMENTED** [review]
Thank you for your contribution. Looking forward to building Karmada together in the future!
There are some minor suggestions, everything else is ok for me.

File: test/e2e/search_test.go [hunk 3]
```diff
@@ -711,4 +717,197 @@ var _ = ginkgo.Describe("[karmada-search] karmada search testing", ginkgo.Ordere
...
+				f = cmdutil.NewFactory(defaultConfigFlags)
+
+				// create a cluster
+				ginkgo.By(fmt.Sprintf("Creating cluster: %s", clusterName), func() {
+					err := createCluster(clusterName, kubeConfigPath, controlPlane, clusterContext)
+					gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+				})
+
+				// create a ResourceRegistry with a non joined cluster name
+				rrName = "rr-" + rand.String(RandomStrLength)
```
Comment (XiShanYongYe-Chang) [inline test/e2e/search_test.go:754]: Put the `rr-` prefix into the constant block in suite_test.go file will be better.

**Comment by XiShanYongYe-Chang** [comment]
Ok, thanks
Ask @RainbowMango to help run the CI test.
/cc @RainbowMango

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks again~ @zhy76 
/lgtm
/approve

**Comment by XiShanYongYe-Chang** [comment]
Uh-oh, there seems to be a conflict in the PR. We may need to resolve it.

```
$ git fetch upstream
$ git rebase upsmaster/master
# hanlde conflict
$ git add .
$ git rebase --continue
$ git push origin {current-branch} -f 
```

**Comment by zhy76** [comment]
> Uh-oh, there seems to be a conflict in the PR. We may need to resolve it.
> 
> ```
> $ git fetch upstream
> $ git rebase upsmaster/master
> # hanlde conflict
> $ git add .
> $ git rebase --continue
> $ git push origin {current-branch} -f 
> ```

OK,I will handle this.

**Comment by XiShanYongYe-Chang** [comment]
Hi @zhy76, would you like to have a try with another issue #2862 ?

**Pull request merged**